File: app/src/main/java/com/ateneaApp/util/FontCache.java
package com.ateneaApp.util;

import android.content.Context;
import android.graphics.Typeface;

import java.util.Hashtable;

public class FontCache {
    private static FontCache fontCache = new FontCache();
    private final Hashtable<String, Typeface> mCache = new Hashtable<>();

    private FontCache() {
    }

    public static FontCache getInstance() {
        return fontCache;
    }

    public Typeface getFont(Context context, String str) {
        if (!this.mCache.containsKey(str)) {
            this.mCache.put(str, Typeface.createFromAsset(context.getAssets(), str));
        }
        return this.mCache.get(str);
    }
}

File: app/src/main/java/com/ateneaApp/util/FontCache.kt
package com.ateneaApp.util

import android.content.Context
import android.graphics.Typeface
import java.util.*

class FontCache {

    private val mCache: Hashtable<String, Typeface> = Hashtable()
    private val fontCache = FontCache()

    fun getInstance(): FontCache {
        return fontCache
    }

    fun getFont(context: Context, str: String?): Typeface? {
        if (!mCache.containsKey(str)) {
            mCache[str] = Typeface.createFromAsset(context.assets, str)
        }
        return mCache[str]
    }
}

